tutorials/13_diagnosis_sbc.ipynb

```diff
@@ -0,0 +1,1897 @@
+    "from sbi.analysis import check_sbc, run_sbc, get_nltp\n",
+    "from sbi.inference import SNPE, SNPE_C, prepare_for_sbi, simulate_for_sbi\n",
+    "from sbi.simulators import linear_gaussian, diagonal_linear_gaussian\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "num_dim = 2\n",
+    "num_simulations = 5_000\n",
+    "\n",
+    "prior_mean = ones(num_dim)\n",
+    "prior_cov = 2*eye(num_dim)\n",
+    "prior = MultivariateNormal(loc=prior_mean, covariance_matrix=prior_cov, validate_args=False)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "collapsed": false
+   },
+   "source": [
+    "## An ideal case\n",
+    "\n",
+    "To explore SBC, we make our life easy and assume that we deal with a problem where the likelihood is modelled by an identity mapping and a bit of smear. But to start, we only use an almost vanishing smear of `0.01`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [
+    {
+     "data": {
+      "application/vnd.jupyter.widget-view+json": {
+       "model_id": "d7b47819a29b4fc6a19dd624b53137a4",
+       "version_major": 2,
+       "version_minor": 0
+      },
+      "text/plain": [
+       "Running 5000 simulations.:   0%|          | 0/5000 [00:00<?, ?it/s]"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "\n",
+    "default_likelihood_loc = 0. #let's start with 0 shift\n",
+    "default_likelihood_scale = 0.01 #let's smear theta only by a little bit\n",
+    "\n",
+    "def simulator(theta, loc = default_likelihood_loc, scale = default_likelihood_scale):\n",
+    "    \"\"\" linear gaussian inspired by sbibm\n",
+    "    s://github.com/sbi-benchmark/sbibm/blob/15f068a08a938383116ffd92b92de50c580810a3/sbibm/tasks/gaussian_linear/task.py#L74\n",
+    "    \"\"\"\n",
+    "    num_dim = theta.shape[-1]\n",
+    "    cov_ = scale*eye(num_dim) #always positively semi-definite\n",
+    "\n",
+    "    # using validate_args=False disables sanity checks on `covariance_matrix`\n",
+    "    # for the sake of speed\n",
+    "    value = MultivariateNormal(loc=(theta+loc), covariance_matrix=cov_, validate_args=False).sample()\n",
+    "    return value\n",
+    "\n",
+    "simulator, prior = prepare_for_sbi(simulator, prior)\n",
+    "theta, x = simulate_for_sbi(simulator, prior, num_simulations)\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "theta: [0.68821174 4.186863  ]\n",
+      "x    : [0.69967127 4.3175235 ]\n"
+     ]
+    }
+   ],
+   "source": [
+    "# we define a hold-out sample of 100 to build an observation\n",
+    "theta_holdout = prior.sample((100,))\n",
```

Comment:
these 100 samples are not used again, are they? maybe just draw one random sample for the illustration below?

Reply:
a remnant from experimentation. Well spotted. a5fa93a